src/core/ext/filters/client_channel/lb_policy/round_robin/round_robin.c

```diff
@@ -630,13 +639,58 @@ static void rr_connectivity_changed_locked(grpc_exec_ctx *exec_ctx, void *arg,
         GRPC_CLOSURE_SCHED(exec_ctx, pp->on_complete, GRPC_ERROR_NONE);
         gpr_free(pp);
       }
+      rr_subchannel_list_unref(exec_ctx, sd->subchannel_list,
+                               "sd_shutdown+started_picking");
+      // unref the "rr_connectivity_update" weak ref from start_picking.
+      GRPC_LB_POLICY_WEAK_UNREF(exec_ctx, &p->base,
+                                "rr_connectivity_sd_shutdown");
+    } else {
+      // Policy isn't shutting down, we aren't in connectivity shutdown (from
```

Comment:
I think the shutdown handling is still wrong here.  If a subchannel's connectivity goes to SHUTDOWN, then we will increment the number of shutdown subchannels above.  We then recreate the subchannel, but we are not decrementing the number of shutdown subchannels.  Won't this cause that number to grow without bound?

Also, consider what happens if we have only one subchannel and it reports as shutdown.  Don't we need to avoid shutting down the RR policy if we're going to try to reconnect?